gcs/src/main/scala/quasar/blobstore/gcs/GCSListService.scala

```diff
@@ -101,19 +135,42 @@ object GCSListings {
       },{j => {
         val items = (j --\ "items").either
         val prefixes = (j --\ "prefixes").either
-        (items, prefixes) match {
-          case (Left(_), Left(_)) => DecodeResult.ok(GCSListings(List.empty[GCSFile]))
-          case (Left(_), Right(p)) => for {
+        val token = (j --\ "nextPageToken").either
+
+        //TODO: there's probably a better way to do this
+        (items, prefixes, token) match {
```

Comment:
and then you don't have to include `token` in the match:
```
        (items, prefixes) match {
          case (Left(_), Left(_)) => for {
            pageToken <- token
          } yield GCSListings(List.empty[GCSFile], pageToken)
          case (... // similar for other cases
```

It should also be possible to further cleanup by creating a codec for the `items` / `prefixes` combi, but I'm fine leaving this as-is too.